### PR TITLE
Manual approval of registration requests

### PR DESCRIPTION
**Pull request by nadinequinr3**
Preview currently building to https://nadine.preview.docs.r3.com/en/platform/corda/5.0-beta/operating/manual-registration-approval.html
(Powershell commands will need to be added in a future PR.)

### CODE REVIEW

File: content/en/platform/corda/5.0-beta/operating/manual-registration.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    parent: corda-5-beta-operate
+    weight: 1000
+section_menu: corda-5-beta
+---
+
+Membership groups can be configured to specify that the MGM operator must manually approve (or decline) member registration requests. 
+The approval process can specify that requests satisfying specific criteria require manual approval, while others are approved automatically. 
+The manual approval process presents the request to the MGM operator, enabling the operator to review the request before approving or declining it using the REST API. 
+This process applies to both registration and re-registration requests. 
+The approval process can be configured at any point in time, and only affects future registration requests - previously approved members are not required to re-register.
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/5.0-beta/operating/manual-registration.md:16]:
```suggestion
The approval process can be configured at any point in time, and only affects future registration requests: previously approved members are not required to re-register.
```

File: content/en/platform/corda/5.0-beta/operating/manual-registration.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+Registration requests are evaluated according to regular expression-based rules defined by the MGM operator. 
+The proposed [MemberInfo]({{< relref "../developing/api/api-membership.md#memberinfo" >}}) is compared with the previous (if any) `MemberInfo` to calculate the difference in their member contexts.
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/5.0-beta/operating/manual-registration.md:19]: Should MemberInfo here use the same formatting in both cases?

File: content/en/platform/corda/5.0-beta/operating/operating-tutorials/manual-approval.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+For example, to specify that all requests in the `corda` namespace must be manually approved:
+
+```bash
+RULE_PARAMS='{"ruleParams":{"ruleRegex": "corda.*", "ruleLabel": "Review all changes to keys in the Corda namespace"}}'
+curl --insecure -u <username>:<password> -d "$RULE_PARAMS" <REST-API-URL>/mgm/<MGM-HOLDING-ID>/approval/rules
+```
+
+## Viewing Current Group Approval Rules
+
+To retrive all applied group approval rules, use the [GET method of the
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/5.0-beta/operating/operating-tutorials/manual-approval.md:26]:
```suggestion
To retrieve all applied group approval rules, use the [GET method of the
```

File: content/en/platform/corda/5.0-beta/operating/operating-tutorials/reviewing-manual-approvals.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+```bash
+curl --insecure -u <username>:<password> -X POST <REST-API-URL>/mgm/<MGM-HOLDING-ID>/approve/<REQUEST-ID>
+```
+ {{< note >}}
+You can only approve requests that are in the `PENDING_MANUAL_APPROVAL` status.
+{{< /note >}}
+
+## Declining a Request
+
+To decline a rquest, pass the ID of the request and a reason to the [POST method of the
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/5.0-beta/operating/operating-tutorials/reviewing-manual-approvals.md:57]:
```suggestion
To decline a request, pass the ID of the request and a reason to the [POST method of the
```

File: content/en/platform/corda/5.0-beta/operating/operating-tutorials/reviewing-manual-approvals.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    },
+    "reason": "string",
+    "registrationId": "string",
+    "registrationSent": "2022-06-24T10:15:30Z",
+    "registrationStatus": "SENT_TO_MGM",
+    "registrationUpdated": "2022-06-24T10:15:30Z"
+  }
+```
+
+Requests that are pending manual approval have the status, `PENDING_MANUAL_APPROVAL`.
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/5.0-beta/operating/operating-tutorials/reviewing-manual-approvals.md:35]:
```suggestion
Requests that are pending manual approval have the status `PENDING_MANUAL_APPROVAL`.
```

File: content/en/platform/corda/5.0-beta/operating/operating-tutorials/reviewing-manual-approvals.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+To view requests from a specific member, (for example, `C=GB, L=London, O=Alice`) and include historic requests:
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/5.0-beta/operating/operating-tutorials/reviewing-manual-approvals.md:37]:
```suggestion
To view requests from a specific member (for example, `C=GB, L=London, O=Alice`) and include historic requests:
```

File: content/en/platform/corda/5.0-beta/operating/operating-tutorials/reviewing-manual-approvals.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+## Declining a Request
+
+To decline a rquest, pass the ID of the request and a reason to the [POST method of the
+mgm/{holdingidentityshorthash}/decline/{requestid} endpoint](../../rest-api/C5_OpenAPI.html#tag/MGM-API/operation/post_mgm__holdingidentityshorthash__decline__requestid_). You can retrieve the ID of a request from the response of the GET endpoint described in [Viewing Requests Pending Manual Approval]({{< relref "#viewing-requests-pending-manual-approval" >}}). 
+
+```bash
+curl --insecure -u <username>:<password> -d <REASON> <API-URL>/mgm/<MGM-HOLDING-ID>/decline/<REQUEST-ID>
+```
+ {{< note >}}
+You can only decline requests that are in the `PENDING_MANUAL_APPROVAL` status.
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/5.0-beta/operating/operating-tutorials/reviewing-manual-approvals.md:64]:
```suggestion
You can only decline requests that have the status `PENDING_MANUAL_APPROVAL`.
```

File: content/en/platform/corda/5.0-beta/operating/manual-registration.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
+---
+date: '2023-03-15'
+title: "Manual Registration Approval"
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/5.0-beta/operating/manual-registration.md:3]: General comment on this and other files; should the filename when created match the title?

**Review by paulmoloneyr3 — CHANGES_REQUESTED** [review]

File: content/en/platform/corda/5.0-beta/operating/manual-registration.md (hunk 2 above)
Comment (nadinequinr3) [reply content/en/platform/corda/5.0-beta/operating/manual-registration.md:19]: I actually logged this to discuss in the next team sync.
Do we have a standard for links with code?

File: content/en/platform/corda/5.0-beta/operating/manual-registration.md (hunk 8 above)
Comment (nadinequinr3) [reply content/en/platform/corda/5.0-beta/operating/manual-registration.md:3]: Yes. I was chopping and changing titles and should have come back to this. Thanks.

File: content/en/platform/corda/5.0-beta/operating/operating-tutorials/configuring-manual-approval-rules.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    weight: 1000
+section_menu: corda-5-beta
+---
+As described in [Manual Registration Approval]({{< relref "../manual-registration.md" >}}), membership groups can require that member registration requests are manually approved (or declined). This section describes how to manage manual registration rules.
+
+## Adding a Group Approval Rule
+
+To add a group approval rule, use the [PUT method of the mgm/{holdingidentityshorthash}/approval/rules endpoint](../../rest-api/C5_OpenAPI.html#tag/MGM-API/operation/post_mgm__holdingidentityshorthash__approval_rules) of the REST API.
+
+For example, to specify that all requests in the `corda` namespace must be manually approved:
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-beta/operating/operating-tutorials/configuring-manual-approval-rules.md:17]: @YashNabar This wasn't very clear in your draft. Can you be sure to review? Thanks

**Review by yarivus — COMMENTED** [review]
LGTM

File: content/en/platform/corda/5.0-beta/operating/operating-tutorials/configuring-manual-approval-rules.md (hunk 9 above)
Comment (YashNabar) [reply content/en/platform/corda/5.0-beta/operating/operating-tutorials/configuring-manual-approval-rules.md:17]: Perhaps something along the lines of
```suggestion
For example, to specify that all requests with changes to keys in the 'corda' namespace must be manually approved:
```

**Review by yarivus — APPROVED** [review]
LGTM

**Review by YashNabar — APPROVED** [review]

File: content/en/platform/corda/5.0-beta/operating/manual-registration.md (hunk 2 above)
Comment (paulmoloneyr3) [reply content/en/platform/corda/5.0-beta/operating/manual-registration.md]: Can it be a link within code bars?

Comment (nadinequinr3) [reply content/en/platform/corda/5.0-beta/operating/manual-registration.md]: No, that doesn't work currently 
image

**Review by paulmoloneyr3 — APPROVED** [review]

**Pull request merged**